W-18904205 mulesoft inference 1.0 (#2935)
mulesoft/docs-connectors: mulesoft-inference/1.0/modules/ROOT/pages/index.adoc

= MuleSoft Inference Connector 1.0 
:lc-connector-name: mulesoft-inference

> Anypoint Connector for MuleSoft Inference (MuleSoft Inference Connector) provides access to inference offerings for large language models (LLMs) from multiple providers including OpenAI, NVIDIA NIM, Perplexity, Anthropic, and many others. Use MuleSoft Inference Connector to easily build and manage AI-driven agents within Anypoint Platform. MuleSoft Inference Connector provides operations to interface directly with the API of various inference providers, enabling seamless integration of AI capabilities into your Mule applications.

Reviewer: I'd recommend we remove this sentence "Use MuleSoft Inference Connector to easily build and manage AI-driven agents within Anypoint Platform."

Reviewer: Anypoint Connector for MuleSoft Inference (MuleSoft Inference Connector) provides access to inference offerings for large language models (LLMs) from multiple providers including OpenAI, NVIDIA NIM, Perplexity, Anthropic, and many others. The connector provides operations to interface directly with the API of various inference providers, enabling seamless integration of AI capabilities into your Mule applications.

Reviewer 2: For the LLMs listed in this sentence, should we mention the ones we are doing GA for now..

For information about compatibility and fixed issues, see the MuleSoft Inference Connector release notes.

== Before You Begin

To use this connector, you must be familiar with:

* Anypoint Connectors
* Mule runtime engine (Mule)
* Elements and global elements in a Mule flow
* How to create a Mule app using Anypoint Code Builder or Anypoint Studio

Before creating an app, you must have:

* Java 17 (required for compilation and runtime)
* Apache Maven
* Credentials to access the MuleSoft Inference Connector target resource
* Anypoint Platform
* The latest versions of Anypoint Code Builder or Anypoint Studio

== Key Features

MuleSoft Inference Connector simplifies AI integration into Mule applications with:

* *Seamless Interaction with Inferenced LLMs*
+

> Integrate and utilize large language models (LLMs) effortlessly for tasks such as natural language processing, text generation, and more, including advanced features such as text generation, analysis, and complex language tasks.

Reviewer: Text generation probably shouldn't be mentioned here twice. 

I'd argue some of the advanced features are image analysis/generation + detection of toxicity in both user provided and LLM generation text.

* *Optimized Performance*
+
Deliver high efficiency and performance in enterprise-grade Mule apps, ensuring smooth handling of AI operations.

> * *HTTPS Security*

Reviewer: This looks like it meant to be expanded upon...

Perhaps it should be: 

**Secure Integration**
Connect corporate applications and data with innovative new AI offerings without damaging security posture.

+
Provide full xref:mule-runtime::tls-configuration.adoc[TLS] support for secure Mule apps.

== Supported Inference Providers

> * AI21Labs (beta)

Reviewer: Are we including the definition of beta support vs. not somewhere in the documentation?

Author: What is the definition? I can add it @rstastny9

* Anthropic (beta)
* Azure AI Foundry
* Azure OpenAI
* Cerebras (beta)
* Cohere (beta)

> * Databricks

Reviewer 2: DataBricks is not GA. Please validate the complete list
The current list of GA connectors includes only:
_Azure OpenAI
Azure DataFoundry
Heroku
NVIDIA
OpenAI
OpenAI Compatible
OpenRouter_

* DeepInfra (beta)
* DeepSeek (beta)
* Docker Models (beta)
* Fireworks (beta)
* GitHub Models (beta)
* Google Vertex AI Express (beta)
* GPT4ALL (beta)
* Groq AI (beta)
* Heroku AI
* Hugging Face (beta) (`hf-inference` only)
* LM Studio (beta)
* Mistral AI (beta)
* NVIDIA 
* Ollama (beta)
* OpenAI
* OpenAI Compatible Endpoints
* OpenRouter
* Perplexity (beta)
* Portkey (beta)
* Together.ai (beta)
* XAI (beta)
* Xinference (beta)
* ZHIPU AI (beta)

== Supported Moderation Providers

* Mistral AI (beta)
* OpenAI

== Supported Vision Model Providers

* Anthropic (beta)
* Azure AI Foundry
* GitHub Models (beta)
* Google Vertex AI Express (beta)
* Groq AI (beta)
* Hugging Face (beta) (`hf-inference` only)
* Mistral AI (beta)
* Ollama (beta)
* OpenAI

> * OpenAI Compatible Endpoints

Reviewer 2: This does not support vision operations. Needs to be removed

* OpenRouter
* Portkey (beta)
* XAI (beta)

== Supported Image Generation Providers

* Heroku AI
* Hugging Face (beta) (`hf-inference` only)
* OpenAI
* Stability AI (beta)
* XAI (beta)

== Supported Operations by Offering

Not all operations are supported by each provider. This table provides a detailed view of operation support across all providers.

> [%header,cols="1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1"]

Reviewer 2: operations are limited while the LLM inferences are expandable. So can we have LLM names in vertical rows and operations as horizontal column headers? @sathya0

|===

> |Operation |GitHub Models |Groq AI |Hugging Face |Ollama |OpenAI |Azure OpenAI |XAI |OpenRouter |Portkey |Perplexity |Cerebras |NVIDIA |Together.ai |Fireworks |DeepInfra |Mistral AI |Anthropic |AI21Labs |Cohere |Xinference |Azure AI Foundry |Google Vertex AI (Gemini) |GPT4ALL |LM Studio |Docker Models |DeepSeek |Zhipu |Data Bricks |Stability AI |Heroku AI

Reviewer 2: Also for the list of what operations are supported by inferences, please refer this sheet only for this:
https://docs.google.com/spreadsheets/d/1NmtzfgU_4YaNmdXgozsCPEwX8zKsWjAFcOk-yBq6sXg/edit?gid=0#gid=0

= MuleSoft Inference Connector 1.0 
:lc-connector-name: mulesoft-inference

> Anypoint Connector for MuleSoft Inference (MuleSoft Inference Connector) provides access to inference offerings for large language models (LLMs) from multiple providers including Groq, Hugging Face, GitHub Models, OpenAI, Anthropic, and many others. Use MuleSoft Inference Connector to easily build and manage AI-driven agents within Anypoint Platform. MuleSoft Inference Connector provides operations to interface directly with the API of various inference providers, enabling seamless integration of AI capabilities into your Mule applications.

Reviewer 3: ```suggestion
Anypoint Connector for MuleSoft Inference (MuleSoft Inference Connector) provides access to inference offerings for large language models (LLMs) from multiple providers including OpenAI, NVIDIA NIM, Perplexity, Anthropic, and many others. Use MuleSoft Inference Connector to easily build and manage AI-driven agents within Anypoint Platform. MuleSoft Inference Connector provides operations to interface directly with the API of various inference providers, enabling seamless integration of AI capabilities into your Mule applications.
```

For information about compatibility and fixed issues, see the MuleSoft Inference Connector release notes.

== Before You Begin

> To use this connector, you must be familiar with:

Reviewer 3: Ayush to review

* Anypoint Connectors
* Mule runtime engine (Mule)
* Elements and global elements in a Mule flow
* How to create a Mule app using Anypoint Code Builder or Anypoint Studio

Before creating an app, you must have:

> * Java 17 (required for compilation and runtime)

Reviewer 3: ```suggestion
* Java 17 (required for compilation and runtime)
Min runtime requirement: 4.9.4
```

Author: This will be displayed in the right nav

* Apache Maven
* Credentials to access the MuleSoft Inference Connector target resource
* Anypoint Platform
* The latest versions of Anypoint Code Builder or Anypoint Studio

== Key Features

MuleSoft Inference Connector simplifies AI integration into Mule applications with:

* *Seamless Interaction with Inferenced LLMs*
+
Integrate and utilize large language models (LLMs) effortlessly for tasks such as natural language processing, text generation, and more, including advanced features such as text generation, analysis, and complex language tasks.
* *Optimized Performance*
+
Deliver high efficiency and performance in enterprise-grade Mule apps, ensuring smooth handling of AI operations.
* *Comprehensive AI Tools and Services*
+

> Access a wide array of AI-driven features, including Retrieval-augmented generation (RAG) for document retrieval, and dynamic tool integration (function calling) for tasks such as recognition and manipulation.

Reviewer 3: ```suggestion
```

> * *MCP Server Support*

Reviewer 3: ```suggestion

```

Reviewer 3: we wont be adding MCP in this version

Author: @nvp008 We won't be adding MCP server, but will we still have the MCP tools operation?

+
Support any MCP servers deployed in your ecosystem through an integrated MCP client. MCP servers must support SSE over HTTP.
* *HTTPS Security*
+
Provide full xref:mule-runtime::tls-configuration.adoc[TLS] support for secure Mule apps.

== Supported Inference Providers

* AI21Labs
* Anthropic
* Azure AI Foundry
* Azure OpenAI
* Cerebras
* Cohere
* Databricks
* DeepInfra
* DeepSeek
* Docker Models
* Fireworks
* GitHub Models

> * Google Vertex AI

Reviewer 3: ```suggestion
* Vertex AI express 
```

* GPT4ALL
* Groq AI
* Hugging Face

> * IBM Watson

Reviewer 3: ```suggestion
* 
```

* LLM API
* LM Studio
* Mistral
* NVIDIA
* Ollama
* OpenAI
* OpenAI Compatible Endpoints
* OpenRouter
* Perplexity
* Portkey
* Together.ai
* XAI
* Xinference

> * ZHIPU AI

Reviewer 3: add Heroku AI

== Supported Moderation Providers

* Mistral AI
* OpenAI

== Supported Vision Model Providers

* Anthropic
* Azure AI Foundry
* GitHub Models

> * Google Vertex AI

Reviewer 3: ```suggestion
* 
```

* How to create a Mule app using Anypoint Code Builder or Anypoint Studio

Before creating an app, you must have:

* Java 17 (required for compilation and runtime)
* Apache Maven
* Credentials to access the MuleSoft Inference Connector target resource
* Anypoint Platform
* The latest versions of Anypoint Code Builder or Anypoint Studio

== Key Features

MuleSoft Inference Connector simplifies AI integration into Mule applications with:

* *Seamless Interaction with Inferenced LLMs*
+
Integrate and utilize large language models (LLMs) effortlessly for tasks such as natural language processing, text generation, and more, including advanced features such as image analysis or generation, and toxicity detection in both user-provided and LLM-generated text.
* *Optimized Performance*
+
Deliver high efficiency and performance in enterprise-grade Mule apps, ensuring smooth handling of AI operations.
* *Secure Integration*
+
Connect corporate applications and data with innovative new AI offerings without damaging security posture. Provide full xref:mule-runtime::tls-configuration.adoc[TLS] support for secure Mule apps.

== Supported Inference Providers

> 

Reviewer: @sathya0 I'm thinking it makes more sense to move the beta disclaimer here or right before the operations table versus up top where you currently have it
